src/components/Resource/ResourceCreate.tsx

```diff
@@ -104,40 +102,32 @@ export default function ResourceCreate(props: ResourceProps) {
     },
   });
 
-  const onSubmit = async (data: ResourceFormValues) => {
-    setIsLoading(true);
-
-    try {
-      const resourceData: CreateResourceRequest = {
-        status: "PENDING",
-        category: data.category,
-        origin_facility: String(props.facilityId),
-        assigned_facility: data.assigned_facility?.id || null,
-        approving_facility: null,
-        emergency: data.emergency === "true",
-        title: data.title,
-        reason: data.reason,
-        referring_facility_contact_name: data.referring_facility_contact_name,
-        referring_facility_contact_number:
-          data.referring_facility_contact_number,
-        related_patient: related_patient,
-        priority: data.priority,
-      };
-
-      const { res, data: responseData } = await request(routes.createResource, {
-        body: resourceData,
-      });
-
-      if (res?.ok && responseData) {
-        toast.success(t("resource_created_successfully"));
-        navigate(`/facility/${facilityId}/resource/${responseData.id}`);
-      }
-    } catch (error) {
-      console.error(error);
+  const { mutate: createResource, isPending } = useMutation({
+    mutationFn: mutate(routes.createResource),
+    onSuccess: (data: ResourceRequest) => {
+      toast.success(t("resource_created_successfully"));
+      navigate(`/facility/${facilityId}/resource/${data.id}`);
+    },
+    onError: (_error) => {
       toast.error(t("something_went_wrong"));
-    } finally {
-      setIsLoading(false);
-    }
+    },
+  });
+
+  const onSubmit = async (data: ResourceFormValues) => {
```

Comment:
are you sure this needs to be async?

```suggestion
  const onSubmit = (data: ResourceFormValues) => {
```